src/tr1d1um/http.go

```diff
@@ -20,107 +17,121 @@ type ConversionHandler struct {
 	timeOut     time.Duration
 	targetUrl   string
 	/*These functions should be set during handler set up */
-	GetFlavorFormat     func(*http.Request, string, string, string) ([]byte, error)
-	SetFlavorFormat     func(*http.Request, BodyReader) ([]byte, error)
-	DeleteFlavorFormat  func(Vars, string) ([]byte, error)
-	AddFlavorFormat     func(io.Reader, Vars, string, BodyReader) ([]byte, error)
-	ReplaceFlavorFormat func(io.Reader, Vars, string, BodyReader) ([]byte, error)
+	GetFlavorFormat     func(*http.Request, string, string, string) (*GetWDMP, error)
+	SetFlavorFormat     func(*http.Request) (*SetWDMP, error)
+	DeleteFlavorFormat  func(Vars, string) (*DeleteRowWDMP, error)
+	AddFlavorFormat     func(io.Reader, Vars, string) (*AddRowWDMP, error)
+	ReplaceFlavorFormat func(io.Reader, Vars, string) (*ReplaceRowsWDMP, error)
+
+	SendRequest    func(*ConversionHandler, http.ResponseWriter, *wrp.Message)
+	HandleResponse func(*ConversionHandler, http.ResponseWriter, *http.Request)
+	EncodeJson     func(interface{}) ([]byte, error)
 }
 
-func (sh ConversionHandler) ConversionGETHandler(resp http.ResponseWriter, req *http.Request) {
-	wdmpPayload, err := sh.GetFlavorFormat(req, "attributes", "names", ",")
+func (sh *ConversionHandler) ConversionHandler(resp http.ResponseWriter, req *http.Request) {
+	var err error
+	var wdmp interface{}
 
-	if err != nil {
-		sh.errorLogger.Log(logging.MessageKey(), ERR_UNSUCCESSFUL_DATA_PARSE, logging.ErrorKey(), err.Error())
-		return
-	}
+	switch req.Method {
+	case http.MethodGet:
+		wdmp, err = sh.GetFlavorFormat(req, "attributes", "names", ",")
+		break
 
-	wrpMessage := &wrp.SimpleRequestResponse{Type: wrp.SimpleRequestResponseMessageType, Payload: wdmpPayload}
+	case http.MethodPatch:
+		wdmp, err = sh.SetFlavorFormat(req)
+		break
 
-	sh.SendData(resp, wrpMessage)
-}
+	case http.MethodDelete:
+		wdmp, err = sh.DeleteFlavorFormat(mux.Vars(req), "parameter")
+		break
 
-func (sh ConversionHandler) ConversionSETHandler(resp http.ResponseWriter, req *http.Request) {
-	wdmpPayload, err := sh.SetFlavorFormat(req, ioutil.ReadAll)
+	case http.MethodPut:
+		wdmp, err = sh.ReplaceFlavorFormat(req.Body, mux.Vars(req), "parameter")
+		break
 
-	if err != nil {
-		sh.errorLogger.Log(logging.MessageKey(), ERR_UNSUCCESSFUL_DATA_PARSE, logging.ErrorKey(), err.Error())
-		return
+	case http.MethodPost:
+		wdmp, err = sh.AddFlavorFormat(req.Body, mux.Vars(req), "parameter")
+		break
 	}
```

Comment:
What about other Methods - like OPTIONS?

Reply:
Currently, the mux only allows requests with these methods to get to the conversion handler. We can allow OPTIONS as well and handle here or in another method. What would be the expected handling behavior for OPTIONS?